srcalc/src/main/java/gov/va/med/srcalc/vista/vistalink/VistaLinkProcedureCaller.java

```diff
@@ -200,7 +202,20 @@ private void setArrayParam(
         }
         catch (SecurityIdentityDeterminationFaultException e)
         {
-            throw new IllegalArgumentException("Invalid DUZ", e);
+            /*
+             * Translate the VistALink-specific Exception into a standard Java Exception.
+             * 
+             * There is an AccountNotFoundException, but I'm not sure that
+             * SecurityIdentityDeterminationFaultException is that specific: just use the
+             * generic AccountException.
+             */
+            
+            // AccountNotFoundException doesn't accept a cause, so just log it. (The
+            // underlying SecurityIdentityDeterminationFaultException doesn't provide much
+            // extra information anyway.)
+            LOGGER.info("Translating VistALink exception into an AccountException.", e);
+
+            throw new AccountException("Could not find a VistA user for that DUZ.");
```

Comment:
I found that sometimes finding the right exception when it comes to Vistalink is a bit hard. Good reasoning here.